chalice/awsclient.py

```diff
@@ -75,10 +78,17 @@ def create_function(self, function_name, role_arn, zip_contents):
                 continue
             return response['FunctionArn']
 
-    def update_function_code(self, function_name, zip_contents):
-        # type: (str, str) -> Dict[str, Any]
-        return self._client('lambda').update_function_code(
+    def update_function(self, function_name, zip_contents,
+                        environment_variables=None):
+        # type: (str, str, Optional[Dict[str, str]]) -> Dict[str, Any]
+        lambda_client = self._client('lambda')
+        return_value = lambda_client.update_function_code(
             FunctionName=function_name, ZipFile=zip_contents)
+        if environment_variables is not None:
```

Comment:
Good point, I was hoping to avoid calling update_function_configuration unless needed, I think it's best to just call it every time.  Otherwise I'd have to store previously deployed function config values which I'd prefer not to do (I plan on adding support for all the various lambda config options, just wanted to start with environment_variables to make sure there were no issues)

It is an extra function call, but I don't think it will slow things down too much.

I'll get this updated.